lib/ansible/modules/cloud/azure/azure_rm_dnsrecordset.py

```diff
@@ -174,18 +174,12 @@
 
 from ansible.module_utils.basic import _load_params
 from ansible.module_utils.six import iteritems
-from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase, HAS_AZURE
 
-try:
+if HAS_AZURE:
     from msrestazure.azure_exceptions import CloudError
     from azure.mgmt.dns.models import Zone, RecordSet, ARecord, AaaaRecord, MxRecord, NsRecord, PtrRecord, SrvRecord, TxtRecord, CnameRecord, SoaRecord
-except ImportError:
```

Comment:
What's the effect if here use try/catch

Reply:
`HAS_AZURE` is required below: it seems better to use it there too instead of catching `ImportError` again.

Reply:
In [`azure_rm_common.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/azure_rm_common.py#L113-L135),  `HAS_AZURE` is set to false once some package installed failed. But the install is not limited to those installed here. If all the requirement are satisfied in `azure_rm_common.py`, but some packages are missing here, the application will crash. So here I would like use the try catch instead a conditional check.

Reply:
`azure_rm_common.py` [try to import](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/azure_rm_common.py#L123) `azure.mgmt.dns` and all imports used in [azure_rm_dnsrecordset](https://github.com/ansible/ansible/blob/08e42c6edf4b2e834874c8d267d7dd01a27bbd50/lib/ansible/modules/cloud/azure/azure_rm_dnsrecordset.py#L181) module are provided by `azure.mgmt.dns` since the [first commit](https://github.com/Azure/azure-sdk-for-python/commit/261b165d13a3f24d911a2f47f3869f7ee0dc6177#diff-15ecfeb79d9115533851c26c8350e25f) of `azure.mgmt.dns`.

For example, when using Azure Python package [1.0.0](https://github.com/Azure/azure-sdk-for-python/blob/v1.0.0/azure/setup.py) which requires `azure-mgmt` 0.20.0 which doesn't provide `azure-mgmt-dns` (`azure-mgmt-dns` is included since `azure-mgmt` 0.30.0rc5) `HAS_AZURE`  would be `False`.

Reply:
Since there are some discussion of "how to import azure models", these code may be removed and move the client creation into modules. Besides this story, I don't think the import logic using a conditional check depends on one other check is a good coding style.

Reply:
Done